internal/core/metadata/v2/controller/http/deviceprofile_test.go

```diff
@@ -469,25 +492,159 @@ func TestAddDeviceProfileByYaml_MissingFile(t *testing.T) {
 	assert.Contains(t, res.Message, "missing yaml file")
 }
 
-func createDeviceProfileRequestWithFile(fileContents []byte) (*http.Request, error) {
-	body := new(bytes.Buffer)
-	writer := multipart.NewWriter(body)
-	part, err := writer.CreateFormFile("file", "deviceProfile.yaml")
-	if err != nil {
-		return nil, err
-	}
-	_, err = part.Write(fileContents)
-	if err != nil {
-		return nil, err
+func TestUpdateDeviceProfileByYaml(t *testing.T) {
+	deviceProfileDTO := buildTestDeviceProfileRequest().Profile
+	deviceProfileModel := dtos.ToDeviceProfileModel(deviceProfileDTO)
+	expectedMessage := "Update device profile successfully"
+
+	dic := mockDic()
+	dbClientMock := &dbMock.DBClient{}
+	dbClientMock.On("UpdateDeviceProfile", deviceProfileModel).Return(nil)
+	dic.Update(di.ServiceConstructorMap{
+		v2MetadataContainer.DBClientInterfaceName: func(get di.Get) interface{} {
+			return dbClientMock
+		},
+	})
+
+	controller := NewDeviceProfileController(dic)
+	assert.NotNil(t, controller)
+
+	valid, err := yaml.Marshal(deviceProfileDTO)
+	require.NoError(t, err)
+	req, err := createDeviceProfileRequestWithFile(valid)
+	require.NoError(t, err)
+
+	// Act
+	recorder := httptest.NewRecorder()
+	handler := http.HandlerFunc(controller.UpdateDeviceProfileByYaml)
+	handler.ServeHTTP(recorder, req)
+	var res common.BaseResponse
+	err = json.Unmarshal(recorder.Body.Bytes(), &res)
+
+	// Assert
+	require.NoError(t, err)
+	assert.Equal(t, http.StatusOK, recorder.Result().StatusCode, "HTTP status code not as expected")
+	assert.Equal(t, contractsV2.ApiVersion, res.ApiVersion, "API Version not as expected")
+	assert.Equal(t, http.StatusOK, res.StatusCode, "BaseResponse status code not as expected")
+	assert.Equal(t, expectedMessage, res.Message, "Message not as expected")
+}
+
+func TestUpdateDeviceProfileByYaml_BadRequest(t *testing.T) {
```

Comment:
Add the valid test to this to avoid the duplicated setup code.

Reply:
Move the valid test to this table drive.